tensorly/decomposition/tests/test_candecomp_parafac.py

```diff
@@ -1,10 +1,127 @@
 import numpy as np
-from ..candecomp_parafac import parafac
-from ..candecomp_parafac import non_negative_parafac
+import pytest
+
+from ..candecomp_parafac import (
+    parafac, non_negative_parafac, normalize_factors, initialize_factors)
 from ...kruskal_tensor import kruskal_to_tensor
 from ...random import check_random_state
 from ... import backend as T
 
+
+randomized_test_cases = {
+    'order3_small': {
+        'dimensions': (8, 8, 8), 'rank': 4, 'seed': 0},
+    'order3_large': {
+        'dimensions': (16, 16, 16), 'rank': 12, 'seed': 0},
+    'order3_mixed': {
+        'dimensions': (16, 12, 8), 'rank': 8, 'seed': 0},
+    'order3_small_orthogonal': {
+        'dimensions': (8, 8, 8), 'rank': 4, 'seed': 0, 'orthogonal': True},
+    'order3_large_orthogonal': {
+        'dimensions': (16, 16, 16), 'rank': 12, 'seed': 0, 'orthogonal': True},
+    'order3_mixed_orthogonal': {
+        'dimensions': (16, 12, 8), 'rank': 8, 'seed': 0, 'orthogonal': True},
+    'order4': {
+        'dimensions': (8, 8, 8, 8), 'rank': 4, 'seed': 0},
+    'order4_orthogonal': {
+        'dimensions': (8, 8, 8, 8), 'rank': 4, 'seed': 0, 'orthogonal': True},
+}
+
+
+def create_random_decomposition(dimensions=(10,10,10), rank=10, seed=None,
```

Comment:
This is a duplicate of `tensorly.random.cp_tensor`, the orthogonal option should be added there.
I think it would make more sense to i) sample a random cp tensor and ii) add some noise rather than have it as a parameter in the generation.

Reply:
1. Didn't realize that `random.cp_tensor` existed.`
2. Sure.